### PR TITLE
#26 Fix Jenkins build

### LINKED ISSUE #26: Fix Jenkins Build
- Ensure yarn cache is cleaned to fix sporadic build errors
- Add GitHub back reporting for ESLint build step

### PR DESCRIPTION
**Pull request by ndoschek**
- Remove build retries
- Add code linting as separate pipeline step
- Add clean of yarn cache to fix sporadic build errors
- Add GitHub back reporting for ESLint

Fixes #26

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -58,7 +50,16 @@ pipeline {
...
+                  sh "rm -rf ${YARN_CACHE_FOLDER}"
+                  sh "yarn --ignore-engines --unsafe-perm"
+                }
+            }
+        }
+
+        stage('Codechecks ESLint') {
+            steps {
+                container('node') {
+                    sh "yarn lint -o eslint.xml -f checkstyle"
```
Comment (cdamus) [inline Jenkinsfile:62]: Should a `**/eslint.xml` file be added to `.gitignore`? After running this command locally, I got three XML files that git wants me to add.

**Review by cdamus — APPROVED** [review]
LGTM! And the build options seem to work locally (I don't have an environment for testing the Jenkins configuration).

I do have a question about something to gitignore, but as it would only be an issue for anyone testing the Jenkins build script, I'm not sure that it's important.

File: Jenkinsfile (hunk 1 above)
Comment (ndoschek) [reply Jenkinsfile:62]: Good catch, thanks @cdamus! I added it to `.gitignore` with the updated commit.

**Review by cdamus — APPROVED** [review]
Great! Thanks.

**Pull request merged**